taxonomy-manager-rest-server/src/main/java/com/digirati/taxman/rest/server/taxonomy/ConceptExtractionService.java

```diff
@@ -28,22 +30,13 @@
 
     private TextLookupService textLookupService;
 
-    public ConceptExtractionService() {
-        initialiseExtractionService(DEFAULT_LANGUAGE_KEY, DEFAULT_LANGUAGE_NAME);
-    }
-
-    public void initialiseExtractionService(String languageKey, String languageName) {
-        Stream<ConceptModel> concepts = conceptDao.loadAllRecords().map(record -> {
-            try {
-                return conceptMapper.map(new ConceptDataSet(record));
-            } catch (RdfModelException e) {
-                throw new WebApplicationException(e);
-            }
-        });
+    @PostConstruct
+    public void initialiseExtractionService() {
```

Comment:
This needs to be done on startup, see `DatabaseInitListener` for an example. You'll want to create a `StartupEvent` listener that injects `TextLookupService`.

```java
class AnalysisInitListener {
    @Inject TextLookupService lookupService; // Or whatever needs initialized at boot.

    void onStartup(@Observes(StartupEvent.class) event) {
         logger.log("Initialized text lookup sevice");
    }
}
```

Since the `IdResolvers` are request scoped you'll get an injection error (there's no `RequestScope` during a `StartupEvent`) so you'll have to change that code to support resolving IDs outside of a request. A simple approach is just adding the base URI of the server as a configuration property instead of pulling it from the request and making the bean `ApplicationScoped`.

Reply:
Will do.

Btw, I've already resolved the `IdResolver` issue with the `Concept` interface. By explicitly sharing the contract between the `ConceptRecord` and `ConceptModel` to provide what the engine needs there's no need to map between them so the `IdResolvers` aren't needed as part of this any more.

Reply:
Hm, I don't think bypassing the data model and passing data records straight down is the best solution here, but I don't see any problems it causes. Happy to stick with that.